compiler-utils/src/testFixtures/java/com/squareup/anvil/compiler/internal/testing/AnvilCompilation.kt

```diff
@@ -91,6 +96,17 @@ public class AnvilCompilation internal constructor(
     }
   }
 
+  /**
+   * Allows you to enable or disable Anvil in the compilation. By default Anvil is enabled.
+   */
+  public fun enableAnvil(enable: Boolean) = apply {
+    checkNotCompiled()
+    if (!enable) {
+      check(!anvilConfigured) { "You cannot disable Anvil after it has been configured." }
```

Comment:
I can't help but feel that "configuring" and "enabling" flows could be streamlined. I'm sure they're split for good reason, but it's not clear when would Anvil be configured yet not enabled? If configuration is dependent on it being enabled, why not just enable it instead of doing a check?

Reply:
> I can't help but feel that "configuring" and "enabling" flows could be streamlined. I'm sure they're split for good reason, but it's not clear when would Anvil be configured yet not enabled? If configuration is dependent on it being enabled, why not just enable it instead of doing a check?

I agree. It would be helpful to understand the use-case for disabling Anvil / when would we disable it. Maybe this info could even be included in the doc comment here.

Reply:
Oh yes, that's much better now. 

There is only one use case I see mentioned below :-)